ui/app/scripts/stream/services/layout.js

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2015-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Implement layout function for streams.
+ *
+ * @author Andy Clement
+ * @author Alex Boyko
+ */
+define(function (require) {
+	'use strict';
+	
+    var DEBUG = false;
+    
+    var joint = require('joint');
+    
+    /**
+	 * Graph representation of stream(s)/module(s).
+	 * @type {joint.dia.Graph}
+	 */
+    var graph;
+    
+    function getIncomingLinks(node) {
+        return graph.getConnectedLinks(node,{inbound: true});
+    }
+    
+    function getOutgoingLinks(node,portKind) {
+        var allLinks = graph.getConnectedLinks(node,{outbound: true});
+        var retval = [];
+        // Exclude TAPS
```

Comment:
Is this comment valid still? I don't see we do anything to exclude taps here. Also, if the `portKind` is of `tap` then it would get the links for it I think.